l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -417,47 +465,32 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
             });
             require(proofValid, "yn");
         }
-        require(_amount > 0, "y1");
 
+        require(!_isEraLegacyDeposit(_chainId, _l2BatchNumber, _l2TxNumberInBatch), "ShB: legacy cFD");
         {
-            bool notCheckedInLegacyBridgeOrWeCanCheckDeposit;
-            {
-                // Deposits that happened before the upgrade cannot be checked here, they have to be claimed and checked in the legacyBridge
-                bool weCanCheckDepositHere = !_isEraLegacyDeposit(_chainId, _l2BatchNumber, _l2TxNumberInBatch);
-                // Double claims are not possible, as depositHappened is checked here for all except legacy deposits (which have to happen through the legacy bridge)
-                // Funds claimed before the update will still be recorded in the legacy bridge
-                // Note we double check NEW deposits if they are called from the legacy bridge
-                notCheckedInLegacyBridgeOrWeCanCheckDeposit = (!_checkedInLegacyBridge) || weCanCheckDepositHere;
-            }
-            if (notCheckedInLegacyBridgeOrWeCanCheckDeposit) {
-                bytes32 dataHash = depositHappened[_chainId][_l2TxHash];
-                bytes32 txDataHash = keccak256(abi.encode(_depositSender, _l1Token, _amount));
-                require(dataHash == txDataHash, "ShB: d.it not hap");
-                delete depositHappened[_chainId][_l2TxHash];
+            bytes32 dataHash = depositHappened[_chainId][_l2TxHash];
+            bytes32 assetId;
+            bytes32 txDataHash;
+            if (uint256(_assetId) <= type(uint160).max) {
```

Comment:
I don't think this logic will work. Legacy `claimFailedDeposit` already calls:

```
bytes32 assetId = nativeTokenVault.getAssetId(_l1Asset);
```

before calling this method. In other words, this number should never be less than 2^160. One of the methods needs to be fixed.